### PR TITLE
Remove Keras Dependency from DeepChem

### LINKED ISSUE #345: Keras Models Don't Behave Deterministically with Fixed Seed
Models which use Keras under the hood don't seem to behave deterministically with fixed seed. Such models include the graph-convolutional models and the low-data models. Here are some potential fixes

- For now, we can just report the numbers in the benchmark table with a * attached to indicate the randomness.
- Long run fix is probably to remove Keras dependency so we can control sources of randomness as with the custom Tensorflow models (#296)

@miaecle Have I summarized the issue correctly?

### PR DESCRIPTION
**Pull request by rbharath**
Refactors `deepchem/models/tf_keras_models`. Moves new layers to `dc.nn`. Attempts to minimize Keras dependencies (to fix #345 and #296) by limiting all Keras dependencies to `dc.nn` and explicitly copying Keras code that DeepChem depends on into `dc.nn`.

Will take a few iterations before tests pass.

### CODE REVIEW

**Comment by rbharath** [comment]
Changes in this PR:
- Removes `deepchem/deepchem/models/keras_models/`
- Moves `deepchem/deepchem/models/tensorflow_models/model_ops.py` to `deepchem/nn/model_ops.py`.
- Copies many Keras-TF functions from `keras/keras/backend/tensorflow_backend.py` to `deepchem/nn/model_ops.py`
- Copies many Keras layers from `keras/keras/engine/topology.py` and `keras/keras/layers/core.py` to new files `deepchem/nn/copy.py`. Have stripped out unneeded functionality, like support for masking and support for sparse tensors. Philosophy is to only support functionality needed for DeepChem models.
- Add in `deepchem/nn/activations.py`, `deepchem/nn/initializations.py`, `deepchem/nn/regularizers.py` and `deepchem/nn/constraints.py`. Not sure how much of this will be needed moving forward.
- Remove all keras imports from files and install from `.travis.yml`
- Unfortunately, this now leads in a `six` dependency, but this is pretty lightweight.
